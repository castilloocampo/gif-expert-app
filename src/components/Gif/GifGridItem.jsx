import React from 'react';
import { PropTypes } from 'prop-types';

const GifGridItem = ({title, url}) => {
  return (
    <div data-testid="gif-grid-item"
      className='card animate__animated animate__fadeIn' >
      <img src={url} alt={title+""} />
      <p data-testid="title" >{title}</p>
    </div>
  );
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}

export default GifGridItem;
