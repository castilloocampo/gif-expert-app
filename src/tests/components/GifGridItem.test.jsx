import '@testing-library/jest-dom'
import { render, screen } from "@testing-library/react";
import GifGridItem from '../../components/Gif/GifGridItem';

describe('GifGridItem', () => {
  const gifItem = {
    title: "Un tÃ­tulo",
    url: "http://localhost/algo.png"
  }

  test('should show Correctly', () => {
    const { container } = render(<GifGridItem {...gifItem} />);
    expect(container).toMatchSnapshot();
  });
  
  test('should has paragraph with the title', () => {
    render(<GifGridItem {...gifItem} />);
    expect(screen.getByTestId('title').innerHTML.trim()).toBe(gifItem.title);
    // exp
  });

  test('should be img-source = url prop', () => {
    render(<GifGridItem {...gifItem} />);
    expect(screen.getByRole("img")).toHaveAttribute("src", gifItem.url);
    expect(screen.getByRole("img")).toHaveAttribute("alt", gifItem.title);
  });
  
  test('should has animate__fadeIn class', () => {
    render(<GifGridItem {...gifItem} />);
    expect(screen.getByTestId("gif-grid-item")).toHaveClass("animate__fadeIn");
  });
});
