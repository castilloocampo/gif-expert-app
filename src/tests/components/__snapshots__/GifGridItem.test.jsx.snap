// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGridItem should show Correctly 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
    data-testid="gif-grid-item"
  >
    <img
      alt="Un título"
      src="http://localhost/algo.png"
    />
    <p
      data-testid="title"
    >
      Un título
    </p>
  </div>
</div>
`;
